pkg/virt-handler/non-root.go

```diff
@@ -135,3 +138,58 @@ func (d *VirtualMachineController) nonRootSetup(origVMI, vmi *v1.VirtualMachineI
 	}
 	return nil
 }
+
+func (d *VirtualMachineController) ensureDVsOwnership(vmi *v1.VirtualMachineInstance) error {
+	res, err := d.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return err
+	}
+	passthoughFSVolumes := make(map[string]struct{})
+	for i := range vmi.Spec.Domain.Devices.Filesystems {
+		passthoughFSVolumes[vmi.Spec.Domain.Devices.Filesystems[i].Name] = struct{}{}
+	}
+
+	pvcVolume := make(map[string]v1.VolumeStatus)
+	hotplugVolumes := make(map[string]bool)
+	for _, volumeStatus := range vmi.Status.VolumeStatus {
+		if volumeStatus.HotplugVolume != nil {
+			hotplugVolumes[volumeStatus.Name] = true
+		}
+
+		if volumeStatus.PersistentVolumeClaimInfo != nil {
+			pvcVolume[volumeStatus.Name] = volumeStatus
+		}
+	}
+
+	for i := range vmi.Spec.Volumes {
+		volume := vmi.Spec.Volumes[i]
+		if volumeSource := &vmi.Spec.Volumes[i].VolumeSource; volumeSource.DataVolume != nil {
+			if _, isPassthoughFSVolume := passthoughFSVolumes[volume.Name]; isPassthoughFSVolume {
+				log.Log.V(4).Infof("this volume %s is mapped as a filesystem passthrough, skipping", volume.Name)
+				continue
+			}
+
+			if hotplugVolumes[volume.Name] {
+				log.Log.V(4).Infof("this volume %s is hotplugged, skipping", volume.Name)
+				continue
+			}
+
+			volumeStatus, ok := pvcVolume[volume.Name]
+			if !ok ||
+				volumeStatus.PersistentVolumeClaimInfo.VolumeMode == nil ||
```

Comment:
volumeMode of nil implies filesystem mode. 

```
        // volumeMode defines what type of volume is required by the claim.
        // Value of Filesystem is implied when not included in claim spec.
        // +optional
        VolumeMode *PersistentVolumeMode `json:"volumeMode,omitempty" protobuf:"bytes,6,opt,name=volumeMode,casttype=PersistentVolumeMode"`
```

Reply:
Thanks, note for me I should not copy past :)
https://github.com/kubevirt/kubevirt/blob/main/pkg/host-disk/host-disk.go#L90